### ISSUE TITLE
x/playground: "timeout running program" error on official testable example

### ISSUE THREAD

**Issue by psampaz** [golang/go#39016]
### What version of Go are you using (`go version`)?

<pre>
go1.14.2 on Go playground
</pre>

### Does this issue reproduce with the latest release?
Yes

### What operating system and processor architecture are you using (`go env`)?

<pre>
Go playground
</pre>

### What did you do?

1. I tried to run the testable example on https://golang.org/pkg/net/http/httputil/#DumpResponse

and it failed immediately with the error "timeout running program"

2. I clicked the share button and I was redirected to https://play.golang.org/p/EULBr_niSNP

where the same code run successfully with out any errors 

3. I changed the text in bold: 

w.Header().Set("Date", "**Wed, 19 Jul 1972 19:00:00 GMT**")

with random text just to have different code and clicked run again.

I tried the above multiple times:

Step 1 always fails
Step 2 always runs successfully 
Step 3 sometimes fails and sometimes runs successfully 
 
### What did you expect to see?

Testable examples running successfully in all 3 cases described above

**Comment by psampaz** [comment]
possible related with #38855

**Comment by toothrot** [comment]
I don't believe that #38855 is related to this issue, as that issue is specific to faketime, not to the timeout in running programs. 

I don't see why this program should timeout reliably on the playground, or at all. The running time on the playground on my laptop is around 300ms. I'll see what I can figure out.

**Comment by toothrot** [comment]
There are several places we can write that message. In local testing, it seems like the particular run timeout that is printing it is this one: https://github.com/golang/playground/blob/0298bc862cef82adf756a5d2c5b5b48920b84009/sandbox/sandbox.go#L357

This tells me that the sandbox container has the binary, got it on time, and the binary itself is just taking too long to execute for some reason.

**Comment by psampaz** [comment]
The example in #38855 is this one https://golang.org/pkg/os/exec/#CommandContext and if you try to run it has the same behavior. 

Same for this 

https://golang.org/pkg/net/http/httptest/#Server 

https://golang.org/pkg/net/http/httptest/#NewTLSServer

https://golang.org/pkg/net/http/httputil/#DumpRequest

https://golang.org/pkg/net/http/httputil/#DumpResponse

and I suppose many more

**Comment by toothrot** [comment]
This issue could be due to networking being disabled in sandbox containers in #25224, as even trivial tcp servers don't work on the playground in my testing.

@psampaz It's a nuanced difference, but this issue is not related to #38855. #38855 is caused by someone expecting a timeout inside of their program to occur, which will not happen as expected unless the fake clock is moved forward. On the playground, there is a slightly different version of the Go runtime ("faketime") that implements a fake clock, which helps make snippets repeatable, along with other features that are playground specific. https://golang.org/cl/192740 has the implementation, if you are curious.

This issue, #39016, is the program on the playground timing out, and never finishing execution. It's also a timeout, true. However, it is not a timeout inside the user's program, but a timeout of the playground waiting for the user program to finish.

The server references you posted are helpful, and also confirm that this is a generalized issue with networking in sandboxed containers.

**Comment by toothrot** [comment]
Actually, the fact that it _sometimes_ works leads me to believe it could be an issue with faketime, but certainly not in the same way as reported in #38855.

**Comment by yoursunny** [comment]
Maybe related: a simple program with both `net.Pipe()` and `time.Sleep()` fails in playground with "timeout running program".
Share link: https://play.golang.org/p/IoBh7drjhkT

```go
package main

import (
	"net"
	"time"
)

func main() {
	net.Pipe()
	time.Sleep(time.Millisecond)
}
```

If I delete either line, the problem does not occur.
Despite the error, if I add other code to interact with the `net.Conn` object returned from `net.Pipe()`, they are functional.

**Comment by EduardTsoy** [comment]
It often fails even with the simplest Hello World:

<img src="https://user-images.githubusercontent.com/10813578/108849836-ade68300-75f3-11eb-9513-4afab4fec581.png" width="256">

**Comment by munapower** [comment]
I am getting the same message on the playground, while I am doing the go Tour. I am in the section where they explain Methods 
![image](https://user-images.githubusercontent.com/21000331/108851424-2b45cf80-75c3-11eb-81d4-9565ff0130d2.png)

**Comment by PierreVandeVyvre** [comment]
Same problem on Go Play Space https://goplay.space/ ...

**Comment by gmCrivelli** [comment]
Same problem on the Go Playground tour.

**Comment by LaevusDexter** [comment]
😐

**Comment by LaevusDexter** [comment]
The same with play.golang.org/compile

**Comment by toothrot** [comment]
This was caused by a temporary outage and has been resolved (see #44537).

**Comment by PierreVandeVyvre** [comment]
Thanks very much for your support !
